### PR TITLE
Feature/build view

### PR DESCRIPTION
**Pull request by saied2035**
### Project requirements
- Implement the design from the sneak peek wireframes.
    -  Using my database I implemented wireframes depending on these pictures:
    
blog_root_users blog_user_page blog_user_all_posts blog_single_post

- I used the methods that I created in [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- **Hint:** To see some posts or comments, you might need to add some data to a database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

Thanks for reviewing my PR 😄

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 	<meta charset="utf-8">
 	<title>specific user</title>
 </head>
 <body>
-  <h1>Here are information about specific user</h1>
+<main id ="container"> 
+  <%= render @user%>
+  <%= render partial: "bio" , locals:{bio:@user.bio}%>
+  <%= render partial: "user_information", as: "post" , collection: @user.recent_posts%>
+  <button id="show-posts">See all posts</button>
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb]: `See All Posts` should be a link and opens the user's post index when the user clicks on it.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 <!DOCTYPE html>
 <html>
 <head>
 	<meta charset="utf-8">
 	<title>users</title>
 </head>
 <body>
-  <h1>Here is a list of users</h1>
+  <main id ="container">  
+    <%= render @users %>
```
Comment (SajjadAhmad14) [inline app/views/users/index.html.erb:10]: Please make every user a link on `app/views/users/index.html.erb` and when a user clicks on it the page leads to the user's show page.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
 <!DOCTYPE html>
 <html>
 <head>
 	<meta charset="utf-8">
 	<title>user's posts</title>
 </head>
 <body>
-  <h1>Here is a list of posts for a user</h1> 
+<main id ="container"> 
+  <%= render partial:'users/user', locals:{user: @user} %>
```
Comment (SajjadAhmad14) [inline app/views/posts/index.html.erb:10]: Please use `#recent_comments` to render 5 most recent comments in the `app/views/posts/index.html.erb`.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Saied, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
PR has a title and summary. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by saied2035** [comment]
Hi, @SajjadAhmad14. Thanks a lot! you gave me good notes and I followed what you said. I appreciate your honest review. 😄 I did the changes and I will push it now.

but I still have a question why should we put the recent_comments at the root `app/views/users/index.html.erb`? I put it inside `app/views/users/_post.html.erb` and called it inside `app/views/users/index.html.erb`. I wanted to Know the advantage behind that. 😄 Thanks in advance.

**Comment by andersonlebon** [comment]
Hello @saied2035  I think the idea behind the review of the previous reviewer is to make sure these some recent comments are previewed for each post so that the user will know immediately if there are new comments on a given post before showing that post. That's why the previous reviewer told you to render those 5 recent comments in the `app/views/posts/index.html.erb.` view. in addition to this the  wireframe in requirements shows that those comments should be rendered on the `app/views/posts/index.html.erb.` view page

image

### Anyway, Great job :clap: on implementing the review I can see everything is okay

**Review by andersonlebon — APPROVED** [review]
## Hi @saied2035 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
- [x] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by saied2035** [comment]
Hi @andersonlebon. Thanks for approving and answering my question. I appreciate that a lot. 😄

**Pull request merged**